app/src/org/odk/collect/android/jr/extensions/IntentCallout.java

```diff
@@ -106,15 +104,27 @@ public Intent generate(EvaluationContext ec) {
         return i;
     }
 
-    public boolean processResponse(Intent intent, ODKView currentView, FormInstance instance, File destination) {
+    private void setNodeValue(TreeReference reference, String stringValue) {
+        if (stringValue != null) {
+            EvaluationContext evaluationContext = new EvaluationContext(form.getEvaluationContext(), reference);
```

Comment:
Can we make a note that this code is almost identical to setValueAction's setter and the code below and note that no one should copy this process without refactoring it to unify that code?